frontend/src/metabase/query_builder/actions/core/updateQuestion.ts

```diff
@@ -82,26 +80,22 @@ function getNextTemplateTagEditorState({
   newQuestion: Question;
   isVisible: boolean;
   queryBuilderMode: QueryBuilderMode;
-}): NextTemplateTagEditorState {
-  const currentQuery = currentQuestion?.query() as NativeQuery;
-  const nextQuery = newQuestion.query() as NativeQuery;
-  const previousTags = currentQuery.templateTagsWithoutSnippets?.() || [];
-  const nextTags = nextQuery.templateTagsWithoutSnippets?.() || [];
-
-  if (nextTags.length > previousTags.length) {
-    if (queryBuilderMode !== "dataset") {
-      return "visible";
-    }
-    return nextTags.every(isSupportedTemplateTagForModel)
-      ? "visible"
-      : "hidden";
+}): boolean {
+  // variable tags are not supported by models, so don't change the visibility
+  if (queryBuilderMode === "dataset") {
+    return isVisible;
   }
-
-  if (nextTags.length === 0 && isVisible) {
-    return "hidden";
-  }
-
-  return;
+  const previousTags = currentQuestion?.isNative()
+    ? (currentQuestion.query() as NativeQuery).variableTemplateTags()
+    : [];
+  const nextTags = newQuestion.isNative()
+    ? (newQuestion.query() as NativeQuery).variableTemplateTags()
+    : [];
+  return nextTags.length > previousTags.length
+    ? true
+    : nextTags.length === 0
+    ? false
+    : isVisible;
```

Comment:
From now on we will avoid chained ternaries.

An early `return true` inside an `if (nextTags.length > previousTags.length)` and a ternary at the end should do it.